plutus-ir/test/TestLib.hs

```diff
@@ -1,89 +1,55 @@
+{-# LANGUAGE FlexibleContexts  #-}
 {-# LANGUAGE OverloadedStrings #-}
 module TestLib where
 
 import           Common
+import           PlcTestUtils
+import           PlutusPrelude                hiding ((</>))
 
-import           Language.PlutusCore.Quote
-import           Language.PlutusIR
+import           Control.Exception
+import           Control.Monad.Except
+import           Control.Monad.Reader         as Reader
 
-import           Data.Text.Prettyprint.Doc
+import qualified Language.PlutusCore.DeBruijn as PLC
+import           Language.PlutusCore.Pretty
+import           Language.PlutusIR.Parser     as Parser
 
-goldenPir :: String -> Term TyName Name a -> TestNested
-goldenPir name value = nestedGoldenVsDoc name $ pretty value
+import           System.FilePath              (joinPath, (</>))
 
-maybeDatatype :: Quote (Datatype TyName Name ())
-maybeDatatype = do
-    m <- freshTyName () "Maybe"
-    a <- freshTyName () "a"
-    match <- freshName () "match_Maybe"
-    nothing <- freshName () "Nothing"
-    just <- freshName () "Just"
+import           Text.Megaparsec.Error        as Megaparsec
 
-    pure $
-        Datatype ()
-            (TyVarDecl () m (KindArrow () (Type ()) (Type ())))
-            [
-                TyVarDecl () a (Type ())
-            ]
-        match
-        [
-            VarDecl () nothing (TyApp () (TyVar () m) (TyVar () a)),
-            VarDecl () just (TyFun () (TyVar () a) (TyApp () (TyVar () m) (TyVar () a)))
-        ]
+import qualified Data.Text                    as T
+import qualified Data.Text.IO                 as T
 
-listDatatype :: Quote (Datatype TyName Name ())
-listDatatype = do
-    m <- freshTyName () "List"
-    a <- freshTyName () "a"
-    let ma = TyApp () (TyVar () m) (TyVar () a)
-    match <- freshName () "match_List"
-    nil <- freshName () "Nil"
-    cons <- freshName () "Cons"
+withGoldenFileM :: String -> (T.Text -> IO T.Text) -> TestNested
+withGoldenFileM name op = do
+    filename <- (</> (name ++ ".plc")) <$> currentDir
```

Comment:
The hardcoding of `.plc` does seem unfortunate. I wouldn't hate it if we had to include the extension somewhere in the test specification or something. Or even just drop it.